site/app/models/gradeable/Gradeable.php

```diff
@@ -1902,21 +1910,30 @@ public function getGradingProgress(User $grader) {
                 $sections[$key] = [
                     'total_components' => $value * $num_components,
                     'graded_components' => 0,
+                    'non_late_total_components' => ($value  - (array_key_exists($key, $late_submitted) ? $late_submitted[$key] : 0) ) * $num_components,
```

Comment:
This is a great opportunity to use PHP's [null coalescing](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce) operator:
```suggestion
                    'non_late_total_components' => ($value  - ($late_submitted[$key] ?? 0) ) * $num_components,
```